internal/file/file_manager_service.go

```diff
@@ -303,9 +313,12 @@ func (fms *FileManagerService) ConfigApply(ctx context.Context,
 	if fileErr != nil {
 		return model.RollbackRequired, fileErr
 	}
-
 	// Update map of current files on disk
 	fms.UpdateCurrentFilesOnDisk(files.ConvertToMapOfFiles(fileOverview.GetFiles()))
+	manifestFileErr := fms.UpdateManifestFile(files.ConvertToMapOfFiles(fileOverview.GetFiles()))
```

Comment:
Small thing, could the converting of the map to files be done once 
```suggestion
     files := files.ConvertToMapOfFiles(fileOverview.GetFiles())
     fms.UpdateCurrentFilesOnDisk(files)
	manifestFileErr := fms.UpdateManifestFile(files)
```